### PR TITLE
add theme option

### LINKED ISSUE #13: Make the logo and favicon configurable
as titled, we should also brand them as Fastify

### LINKED ISSUE #35: Swagger themes support
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the feature has not already been requested

### 🚀 Feature Proposal

Integration with different styles can be good for customization. The following repository could be helpful.

https://www.npmjs.com/package/swagger-themes

### Motivation

_No response_

### Example

_No response_

### LINKED ISSUE #38: Custom CSS Theme
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the feature has not already been requested

### 🚀 Feature Proposal

Good time of day. I tried to find an opportunity to connect a custom CSS file (a theme for Swagger), but unfortunately I didn't find it. Is there such a possibility? If not, is it planned?

### Motivation

_No response_

### Example

_No response_

### PR DESCRIPTION
**Pull request by Uzlopak**
Closes #13 
Closes #35
Closes #38

First we should discuss if the approach is OK or not. So basically I introduce a theme option. The theme option can handle additional javascript, additional css files and additional favicons. See the corresponding typescript interface. 
I guess we need also to calculate the CSP of the files on plugin load, which is not yet implemented but should be a no-brainer.

So lets discuss if it is the right approach. 

So for the request of @OkanPinar in #35 you would do:

```js
'use strict'

const fastify = require('fastify')({ logger: true })
const { SwaggerTheme } = require('swagger-themes');

fastify.register(require('@fastify/swagger'), {
  mode: 'static',
  specification: {
    path: './examples/example-static-specification.json'
  }
})
fastify.register(require('../index'), {
  theme: {
    css: [
      { filename: 'theme.css', content: new SwaggerTheme('v3').getBuffer('dark') }
    ],
  }
})

fastify.listen({ port: 3000 }, (err) => {
  if (err) throw err
})
```

Taking the example of @Xpucmoc-Bockpec for #13 

```js
'use strict'

const fastify = require('fastify')({ logger: true })
const css = `h2 + a, section .models, .download-url-wrapper {
  display: none!important;
}

.topbar {
  background-color: #002f6c!important;
}

.opblock-get .opblock-summary-method {
  background-color: #0048a6!important;
}

.topbar-wrapper {
  height: 45px;
  background-repeat: no-repeat;
  background-size: contain;
  background-image: url("hello.svg");
}

.topbar-wrapper img {
  display: none;
}`

fastify.register(require('@fastify/swagger'), {
  mode: 'static',
  specification: {
    path: './examples/example-static-specification.json'
  }
})
fastify.register(require('../index'), {
  theme: {
    css: [
      { filename: 'theme.css', content: css }
    ],
  }
})

fastify.listen({ port: 3000 }, (err) => {
  if (err) throw err
})

```

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by mcollina** [comment]
Can you include one or two examples of this customization?

**Comment by Uzlopak** [comment]
The Logo is an image and that image is the 64x64 favicon. So the answer is yes, but on the other hand also kind of not that "clever" as it is not possible to completely theme the swagger documentation. 

Maybe it makes more sense to have a theme option which basically mounts a folder with fastify-static to `/static/theme`. And we basically expect that the theme folder contains a index.css with the overriding css or maybe index.html overriding everything. Benefit of that would be, that an implementer could put all files into the theme-folder, reducing the configuration overhead and be in full controle of the output.

**Comment by mcollina** [comment]
I think that would be better.

**Comment by Uzlopak** [comment]
Mounting the theme folder seems to be elegant, but then it is not easy to just hook those swagger-themes.

**Comment by mcollina** [comment]
Let's land this and see. Tests are failihg

**Comment by Uzlopak** [comment]
I will probably implement it tomorrow

**Comment by ilyamixaltik** [comment]
As support for the swagger-themes library will be added, please write. I will update the library documentation.

**Comment by ilyamixaltik** [comment]
Are there any successes?

File: examples/theme.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        sizes: '16x16',
+        type: 'image/png',
+        content: Buffer.from('iVBORw0KGgoAAAANSUhEUgAAAA8AAAAQCAQAAABjX+2PAAAAIGNIUk0AAHomAACAhAAA+gAAAIDoAAB1MAAA6mAAADqYAAAXcJy6UTwAAAACYktHRAD/h4/MvwAAAAd0SU1FB+cCEQ06N8A8CiUAAADnSURBVBjTrdE/K8QBAMbxz/0TLrnUWcTg7ySLewGEwWDRzSYpyULJbGG6wWBTlMEbkHsFNnVloAwXudIlnDru1O9nOCex3rM89TzL0/eh1Ypo//Zk5CdM6JP2IWFOxbmMKZVmPWzbrJSamG5FNXUFx42yV16oqCQUerNr2pghsSgS1sw4kxNVVvbu3rwjSwJ67Kgq2XMjtO/AnWsnVgwQNy6rQ8GkURWBpCebXnR5gA11j5b1OxT4EKq6dGurMWvQqqw2LPoUKDq1LqPzN4q0rCuvckbE/pOakHdhQfwvwKan8Nzad74AkR8/Ir6qAvAAAAAldEVYdGRhdGU6Y3JlYXRlADIwMjMtMDItMTdUMTM6NTg6NTUrMDA6MDBjkr64AAAAJXRFWHRkYXRlOm1vZGlmeQAyMDIzLTAyLTE3VDEzOjU4OjU1KzAwOjAwEs8GBAAAACh0RVh0ZGF0ZTp0aW1lc3RhbXAAMjAyMy0wMi0xN1QxMzo1ODo1NSswMDowMEXaJ9sAAAAASUVORK5CYII=', 'base64')
+      }
+    ]
+  }
+})
+
+fastify.listen({ port: 3000 }, (err) => {
+  if (err) throw err
```
Comment (Eomm) [inline examples/theme.js:32]:
```suggestion
  if (err) throw err
  fastify.log.info('Visit the documentation at http://127.0.0.1:3000/documentation/')
```

**Review by Eomm — APPROVED** [review]
Added docs

Great feature 🔝

File: README.md [hunk 2]
```diff
@@ -110,6 +110,7 @@ await fastify.ready()
  | transformSpecificationClone| true             | Provide a deepcloned swaggerObject to transformSpecification                                                                    |
  | uiConfig             | {}               | Configuration options for [Swagger UI](https://github.com/swagger-api/swagger-ui/blob/master/docs/usage/configuration.md).                                                                                                   |
  | uiHooks              | {}               | Additional hooks for the documentation's routes. You can provide the `onRequest` and `preHandler` hooks with the same [route's options](https://www.fastify.io/docs/latest/Routes/#options) interface.|
+ | theme                | {}               | Allow to add custom javascript and css to the Swagger UI web page |
```
Comment (Fdawgs) [inline README.md:113]:
```suggestion
 | theme                | {}               | Add custom JavaScript and CSS to the Swagger UI web page |
```

File: README.md [hunk 3]
```diff
@@ -193,6 +194,37 @@ await fastify.register(require('@fastify/swagger-ui'), {
 })
 ```
 
+#### theme
+
+You can add custom javascript and css to the Swagger UI web page by using the theme option.
```
Comment (Fdawgs) [inline README.md:199]:
```suggestion
You can add custom JavaScript and CSS to the Swagger UI web page by using the theme option.
```

**Review by Fdawgs — CHANGES_REQUESTED** [review]

**Review by Fdawgs — APPROVED** [review]

**Pull request merged**